From 3af08080d48481b6e187bc3319ad900689d7147e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=D0=A0=D1=83=D1=81=D0=BB=D0=B0=D0=BD=20=D0=98=D0=B6=D0=B1?=
 =?UTF-8?q?=D1=83=D0=BB=D0=B0=D1=82=D0=BE=D0=B2?= <lrn1986@gmail.com>
Date: Mon, 13 Apr 2015 12:11:47 +0000
Subject: [PATCH 09/20] [W32] Fix deleteObject prototypte, fix header case

Apparently, <T> is unneeded here:

In file included from ../webkitgtk-2.4.8/Source/WebCore/platform/graphics/win/GraphicsContextCairoWin.cpp:35:0:
../webkitgtk-2.4.8/Source/WTF/wtf/win/GdiObject.h:114:58: error: function template partial specialization 'deleteObject<T>' is not allowed
 template<typename T> inline void deleteObject<T>(T object)
                                                          ^

As a bonus, change GdiObject.h -> GDIObject.h (this matters when
cross-compiling or when using case-sensitive filesystems on W32).
---
 Source/WTF/wtf/win/GDIObject.h                                   | 2 +-
 Source/WebCore/platform/graphics/win/GraphicsContextCairoWin.cpp | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Source/WTF/wtf/win/GDIObject.h b/Source/WTF/wtf/win/GDIObject.h
index e8b00be..4e01b3ce 100644
--- a/Source/WTF/wtf/win/GDIObject.h
+++ b/Source/WTF/wtf/win/GDIObject.h
@@ -111,7 +111,7 @@ template<typename T> inline void swap(GDIObject<T>& a, GDIObject<T>& b)
 }
 
 // Nearly all GDI types use the same DeleteObject call.
-template<typename T> inline void deleteObject<T>(T object)
+template<typename T> inline void deleteObject(T object)
 {
     if (object)
         ::DeleteObject(object);
diff --git a/Source/WebCore/platform/graphics/win/GraphicsContextCairoWin.cpp b/Source/WebCore/platform/graphics/win/GraphicsContextCairoWin.cpp
index 1b2d865..3fb2ac0 100644
--- a/Source/WebCore/platform/graphics/win/GraphicsContextCairoWin.cpp
+++ b/Source/WebCore/platform/graphics/win/GraphicsContextCairoWin.cpp
@@ -32,7 +32,7 @@
 
 #include <cairo-win32.h>
 #include "GraphicsContextPlatformPrivateCairo.h"
-#include <wtf/win/GdiObject.h>
+#include <wtf/win/GDIObject.h>
 
 using namespace std;
 
-- 
1.8.5.3

